### PR TITLE
Fixing (p)read/writev offsets when iovcnt  > 1.

### PR DESCRIPTION
**Pull request by RamiHg**
doReadv and doWritev were not incrementing the offsets when looping over the iovecs. This was causing the read/writes to stomp over the first in the array.

Checking for offset being undefined so that this still works when reading/writing to an unseekable file (e.g. /dev/random).

Also added unit tests.

This one took me quite a while to track down in a large codebase.

Note: I am not a Javascript developer. Please let me know if my change to library_wasi.js can be done in a better way.

### CODE REVIEW

File: src/library_wasi.js [hunk 1]
```diff
@@ -206,11 +207,15 @@ var WasiLibrary = {
       if (curr < 0) return -1;
       ret += curr;
       if (curr < len) break; // nothing more to read
+      if (has_offset) {
```
Comment (sbc100) [inline src/library_wasi.js:210]: Can you make this `if (typeof offset !== 'undefined')` and avoid creating `has_offset` at all?

**Review by sbc100 — COMMENTED** [review]
Nice fine!

File: src/library_wasi.js (hunk 1 above)
Comment (RamiHg) [reply src/library_wasi.js]: Heh, that was my original version - but I wasn't sure if it was considered OK!

File: test/fs/test_writev.c [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  assert(nwritten == 8);
+
+  size_t nread = read(fd, buf0, sizeof(buf0));
+  assert(nread == 8);
+  assert(strcmp(buf0, "a=1\nb=2\n") == 0);
+  close(fd);
+
+  printf("success\n");
+  return 0;
+}
```
Comment (sbc100) [inline test/fs/test_writev.c:34]: Missing newline here

**Review by sbc100 — APPROVED** [review]

**Pull request merged**